test/GroupByTest.cpp

```diff
@@ -1521,16 +1513,14 @@ TEST_F(GroupByOptimizations, computeGroupByForJoinWithFullScan) {
     // Set up a GROUP BY operation for which the optimization can be applied.
     // The last two arguments of the `Join` constructor are the indices of the
     // join columns.
-    IdTable result(qec->getAllocator());
     auto join = makeExecutionTree<Join>(qec, values, xyzScanSortedByX, 0, 0);
     GroupBy validForOptimization{qec, variablesOnlyX, aliasesCountX, join};
-    if (chooseInterface) {
-      ASSERT_TRUE(
-          validForOptimization.computeGroupByForJoinWithFullScan(&result));
-    } else {
-      ASSERT_TRUE(
-          validForOptimization.computeOptimizedGroupByIfPossible(&result));
-    }
+    auto optional =
+        chooseInterface
+            ? validForOptimization.computeGroupByForJoinWithFullScan()
+            : validForOptimization.computeOptimizedGroupByIfPossible();
+    ASSERT_TRUE(optional.has_value());
+    IdTable result = std::move(optional).value();
```

Comment:
You can also simplify the following result checks by the helper function `makeIdTableFromVector` 
(clean up as you go).
This should save us several lines and increase the readability.